app/models/line_item.rb

```diff
@@ -0,0 +1,59 @@
+class LineItem < ApplicationRecord
+
+  ## ASSOCIATINOS
+  belongs_to :order
+  belongs_to :deal
+
+  ## VALIDATIONS
+  # ensuring one quantity of a deal per order
+  validates :quantity, numericality: { 
+    only_integer: true,
+    equal_to: ENV['maximum_number_of_deals_one_can_order'].to_i }
+  
+  # ensure quantity less than or equal to deal.quantity
+  validate :ensure_quantity_less_than_or_equal_to_deals_quantity
+ 
+  validate :ensure_deal_not_bought_again_in_another_order
+
+  ## CALLBACKS
+  before_save :update_loyalty_discount, if: :quantity_changed?
+  after_commit :update_orders_loyalty_discount_and_total_amount
+
+  def total_price
+    quantity * discount_price - loyalty_discount
+  end
+
+  def pretty_error
+    errors.full_messages.join("<br>")
+  end
+
+  private
+
+    def ensure_deal_not_bought_again_in_another_order
+      associated_user = order.user
+      if associated_user.line_items.exists?(deal_id: deal_id) && !order.line_items.exists?(deal_id: deal_id)
```

Comment:
can this be done in one condition
`associated_user.line_items.where(deal_id: deal_id).where.not(order_id: id)`